src/train_tracker.py

```diff
@@ -0,0 +1,228 @@
+"""列車捕捉モジュール.
+
+列車を捕捉する
+@author: desty505 miyashita64
+"""
+
+import cv2
+import numpy as np
+from camera_interface import CameraInterface
+
+
+class TrainTracker:
+    """列車を捕捉するクラス."""
+
+    def __init__(self, camera_id: int) -> None:
+        """コンストラクタ.
+
+        Args:
+            camera_id (int): カメラID
+        """
+        self.camera = CameraInterface(camera_id)
+        self.diff_border = 20
+        self.observe_rect_points = [(0, 0), (0, 0)]
+
+    def __del__(self) -> None:
+        """デストラクタ."""
+        del self.camera
+
+    def calibrate(self) -> None:
+        """キャリブレーション."""
+        # 録画を開始する
+        video, mark_video = self.camera.start_record()
+        # クリックに対するコールバックを設定
+        cv2.namedWindow("Calibration")
+        cv2.setMouseCallback("Calibration", self.mouse_callback)
+
+        while True:
+            # フレームを取得
+            success, frame = self.camera.get_frame()
+            if not success:
+                break
+
+            # クリックされた領域の描画
+            mark_frame = self.draw_observe_rect(frame)
+
+            # フレームの表示
+            cv2.imshow("Calibration", mark_frame)
+            # フレームのファイル出力
+            video.write(frame)              # 撮影した動画
+            mark_video.write(mark_frame)    # 輪郭線を描画した動画
+
+            # "q"キーを押すとループを抜けて終了
+            if cv2.waitKey(1) & 0xFF == ord("q"):
+                break
+
+        # リソースを解放
+        self.camera.end_record()
+
+    def mouse_callback(self, event, x, y, flags, param) -> None:
+        """キャリブレーションのクリック時の処理.
+
+        Args:
+            event: クリックイベント
+            x: クリックのx座標
+            y: クリックのy座標
+            flags: 追加情報のフラグ
+            param: 任意パラメータ
+        """
+        if event == cv2.EVENT_LBUTTONDOWN:
+            # クリック時の座標を記録
+            self.observe_rect_points.append((x, y))
```

Comment:
```suggestion
            # コンストラクタで、self.observe_rect_points = None にして、条件分岐
            if self.observe_rect_points is None:
                self.observe_rect_points = [(x, y)]
            else:
                self.observe_rect_points.append((x, y))

```
動作テストで言ってる「初期値が[(0,0),(0,0)]となっているため、1回目のクリックでは大きな四角形が表示される。」が嫌なら、初期値をNoneにする？個人的に[(0,0),(0,0)]は気持ち悪いけど...

Reply:
「大きな四角が表示される」は、挙動の説明のつもりで問題とは思ってなかった。
最初の1点のために、点の表示処理とか分岐が増えるよりは、初期化する方がいいかと思ったけどどう？